docs/src/development/proposals/MEP9/README.md

```diff
@@ -0,0 +1,66 @@
+# No Open Ports To the Data Center
+
+Our metal-stack partitions typically have open ports for metal-stack native services, these are:
+
+- SSH port on the firewalls
+- bmc-reverse-proxy for serial console access through the metal-console
+
+These open ports are potential security risks. For example, while SSH access is possible only with private key it's still vulnerable to DoS attack.
```

Comment:
We do not have SSH key rotation either, so if somebody leaks a key or leaves the company, this SSH access will probably work for a long time after this incident.

Reply:
With https://github.com/gardener/gardener/pull/4224 SSH Keypair rotation is already implemented in Gardener (>= v1.28.0) But we have not enabled this feature because it would break firewall ssh access once the keys are rotated.

So yes, this is was needs to be done additionally, make the firewall-controller (or maybe another  controller running on the firewall) aware of rotated ssh keys and replace the ssh publickey with the newly created.

Reply:
But the Gardener SSH key rotation is working on cluster nodes through the cloud config downloader. It's a completely separate mechanism from our firewall reconciliation. I don't know how this belongs together?

Reply:
Sure, we need another mechanism on the firewall. Most simple would be polling the ssh pubkey from gardener and compare with the installed.

Reply:
Then this would pretty much fit to the good old https://github.com/metal-stack/firewall-controller/issues/35. If we had a connection to the API server of the seed from the firewall-controller, then we can reconcile the SSH key during our normal reconciliation flow.